tardis/gui.py

```diff
@@ -43,68 +105,739 @@ def __init__(self, parent=None):
+        
+        #Data structures
         self.model = None
         self.shell_info = {}
         self.line_info = []
-        self.setGeometry(20, 35, 1250, 500)
-        self.setWindowTitle('Shells Viewer')
-        self.tablemodel = SimpleTableModel([['Shell: '], ["Rad. temp", "Ws"]], (1, 0))
+
+        #Shells widget
+        self.shellWidget = self.makeShellWidget()
+        
+        #Spectrum widget
+        self.spectrumWidget = self.makeSpectrumWidget()
+
+        #Plot tab widget
+        self.plotTabWidget = QtGui.QTabWidget()
+        self.plotTabWidget.addTab(self.shellWidget,"&Shells")
+        self.plotTabWidget.addTab(self.spectrumWidget, "S&pectrum")
+
+        #Table widget
+        self.tablemodel = SimpleTableModel([['Shell: '], ["Rad. temp", "Ws"]],
+         (1, 0))
         self.tableview = QtGui.QTableView()
+        self.tableview.setMinimumWidth(200)
+        self.tableview.connect(self.tableview.verticalHeader(), 
+            QtCore.SIGNAL('sectionClicked(int)'), self.graph.highlight_shell)
+        self.tableview.connect(self.tableview.verticalHeader(), 
+            QtCore.SIGNAL('sectionDoubleClicked(int)'),
+            self.on_header_double_clicked)
+
+        #Label for text output
+        self.outputLabel = QtGui.QLabel()
+        self.outputLabel.setFrameStyle(QtGui.QFrame.StyledPanel | 
+            QtGui.QFrame.Sunken)
+        self.outputLabel.setStyleSheet("QLabel{background-color:white;}")
+
+        #Group boxes
+        graphsBox = QtGui.QGroupBox("Visualized results")
+        textsBox = QtGui.QGroupBox("Model parameters")
+        tableBox = QtGui.QGroupBox("Tabulated results")
+
+        #For textbox
+        textlayout = QtGui.QHBoxLayout()
+        textlayout.addWidget(self.outputLabel)
+
+        tableslayout = QtGui.QVBoxLayout()
+        tableslayout.addWidget(self.tableview)
+        tableBox.setLayout(tableslayout)
+
+        visualayout = QtGui.QVBoxLayout()
+        visualayout.addWidget(self.plotTabWidget)
+        graphsBox.setLayout(visualayout)
+
+        self.layout = QtGui.QHBoxLayout()
+        self.layout.addWidget(graphsBox)
+        textntablelayout = QtGui.QVBoxLayout()
+        textsBox.setLayout(textlayout)
+        textntablelayout.addWidget(textsBox)
+        textntablelayout.addWidget(tableBox)
+
+        self.layout.addLayout(textntablelayout)                
+        self.setLayout(self.layout)
+
+    def fillOutputLabel(self):
+        labeltext = 'Iterations requested: {} <br/> Iterations executed:  {}<br/>\
+                     Model converged     : {} <br/> Simulation Time    :  {} s <br/>\
+                     Inner Temperature   : {} K <br/> Number of packets  :  {}<br/>\
+                     Inner Luminosity    : {}'\
+                     .format(self.model.iterations_max_requested, 
+                        self.model.iterations_executed,
+                        '<font color="green"><b>True</b></font>' if 
+                        self.model.converged else 
```

Comment:
Well both of your solutions won't work because I added this statement not to change the True to False but the color of the string 'True' or 'False'. False is red and True is green. But in both of your suggested solutions the string will always be green.

@wkerzendorf, if I really want to do a formatting of this string I have to do something like:

```python
    truthstring = "<font color={}>{}</font>"
    if (model.converged):
        truthstring.format('green','True')
    else:
        truthstring.format('red','False')
"blah blah".format("blah1","blah2", truthstring)
```
I think that is needlessly complicated. Will it be more pleasing to the eyes if I do something like:

```python
    truth1 = "<font color=green>True</font>"
    truth2 = "<font color=red>False</font>"
    "blah blah".format("blah1","blah2", truth1 if model.converged else truth2)
```
I still feel like the code already written is the clearest but I don't know what the PEP says about these things. So which one should I put in.